rosidl_cmake/cmake/rosidl_generate_interfaces.cmake

```diff
@@ -133,6 +133,23 @@ macro(rosidl_generate_interfaces target)
   # afterwards all remaining interface files are .idl files
   list(APPEND _idl_tuples ${_idl_adapter_tuples})
 
+  # to generate action interfaces, we need to depend on "action_msgs"
+  foreach(_tuple ${_interface_tuples})
+    string(REGEX REPLACE ".*:([^:]*)$" "\\1" _tuple_file "${_tuple}")
+    get_filename_component(_parent_dir "${_tuple_file}" DIRECTORY)
+    if("${_parent_dir}" STREQUAL "action")
+      find_package(action_msgs QUIET)
+      if(NOT ${action_msgs_FOUND})
+        message(FATAL_ERROR "Unable to generate action interface for '${_tuple_file}'. "
+          "In order to generate action interfaces, you must add a build dependency to 'action_msgs' "
```

Comment:
Not only a build dependency since you also want to export it. So a build and run dependency (`depend`) is necessary.